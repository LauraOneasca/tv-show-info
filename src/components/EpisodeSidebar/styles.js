import { css } from 'styled-components';

export default css`
  background-color: white;
  height: 100%;
  display: flex;
  flex-direction: column;

  @media (min-width: 1024px) {
    width: 37%;
  }

  .sidebar {
    &__image {
      background-repeat: no-repeat;
      background-size: cover;
      width: 100%;
      height: 100%;

      ${({ imageUrl }) =>
        imageUrl &&
        css`
          background-image: url(${imageUrl});
        `}
    }

    &__content {
      min-height: 30rem;
      overflow: hidden;
    }

    &__meta {
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 4rem;
      border-bottom: 0.1rem solid #9797974d;
    }

    &__title {
      font-size: 2.7rem;
      line-height: 3.2rem;
      font-weight: bold;
      margin-bottom: 0.6rem;
    }

    &__info {
      font-size: 1.9rem;
      line-height: 2.2rem;
    }

    &__rating {
      display: flex;
      align-items: center;

      &__star {
        margin-right: 1.7rem;
      }
    }

    &__main-info {
      padding: 4rem;
      text-align: left;
    }
  }
`;
