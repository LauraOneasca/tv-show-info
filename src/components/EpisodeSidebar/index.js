import React from 'react';
import PropTypes from 'prop-types';
import 'styled-components/macro';

import styles from './styles';

/** TODO add real images */
const EpisodeSidebar = ({ info }) => (
  <div css={styles} imageUrl={`https://picsum.photos/620/400?random=${parseInt(info.Episode, 10)}`}>
    <div className="sidebar__image" />
    <div className="sidebar__content">
      <div className="sidebar__meta">
        <div className="sidebar__info">{`Episode ${info.Episode} â€” ${info.Released}`}</div>
        <div className="sidebar__rating sidebar__info">
          <svg
            className="sidebar__rating__star"
            width="28"
            height="28"
            viewBox="0 0 28 28"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 0.692261L18.3237 9.45346L27.9924 10.8587L20.9962 17.678L22.6473 27.3077L14 22.7612L5.35276 27.3077L7.00386 17.678L0.00769043 10.8587L9.67639 9.45346L14 0.692261Z"
              fill="#EFD358"
            />
          </svg>
          {`${info.imdbRating}/10`}
        </div>
      </div>
      <div className="sidebar__main-info">
        <div className="sidebar__title">{info.Title}</div>
        <div className="sidebar__info">{info.Plot}</div>
      </div>
    </div>
  </div>
);

EpisodeSidebar.propTypes = {
  info: PropTypes.shape({
    Episode: PropTypes.string,
    Released: PropTypes.string,
    imdbRating: PropTypes.string,
    Title: PropTypes.string,
    Plot: PropTypes.string,
  }).isRequired,
};

export default EpisodeSidebar;
