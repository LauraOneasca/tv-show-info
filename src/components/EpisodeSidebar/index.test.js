import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import EpisodeSidebar from '.';

const info = {
  Title: '401 Unauthorized',
  Released: '06 Oct 2019',
  Episode: '1',
  Plot:
    'During the Christmas season, Elliot and Mr. Robot make their return; Darlene deals with real trouble; Tyrell is bored; Dom becomes paranoid.',
  imdbRating: '9.4',
};

describe('EpisodeSidebar', () => {
  it('should match the snapshot', () => {
    const tree = render(<EpisodeSidebar info={info} />);

    expect(tree.baseElement).toMatchSnapshot();
  });
});
