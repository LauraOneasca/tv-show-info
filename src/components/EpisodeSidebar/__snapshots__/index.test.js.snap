// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeSidebar should match the snapshot 1`] = `
.c0 {
  background-color: white;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .sidebar__image {
  background-repeat: no-repeat;
  background-size: cover;
  width: 100%;
  height: 100%;
  background-image: url(https://picsum.photos/620/400?random=1);
}

.c0 .sidebar__content {
  min-height: 30rem;
  overflow: hidden;
}

.c0 .sidebar__meta {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 4rem;
  border-bottom: 0.1rem solid #9797974d;
}

.c0 .sidebar__title {
  font-size: 2.7rem;
  line-height: 3.2rem;
  font-weight: bold;
  margin-bottom: 0.6rem;
}

.c0 .sidebar__info {
  font-size: 1.9rem;
  line-height: 2.2rem;
}

.c0 .sidebar__rating {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .sidebar__rating__star {
  margin-right: 1.7rem;
}

.c0 .sidebar__main-info {
  padding: 4rem;
  text-align: left;
}

@media (min-width:1024px) {
  .c0 {
    width: 37%;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="sidebar__image"
      />
      <div
        class="sidebar__content"
      >
        <div
          class="sidebar__meta"
        >
          <div
            class="sidebar__info"
          >
            Episode 1 â€” 06 Oct 2019
          </div>
          <div
            class="sidebar__rating sidebar__info"
          >
            <svg
              class="sidebar__rating__star"
              fill="none"
              height="28"
              viewBox="0 0 28 28"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 0.692261L18.3237 9.45346L27.9924 10.8587L20.9962 17.678L22.6473 27.3077L14 22.7612L5.35276 27.3077L7.00386 17.678L0.00769043 10.8587L9.67639 9.45346L14 0.692261Z"
                fill="#EFD358"
              />
            </svg>
            9.4/10
          </div>
        </div>
        <div
          class="sidebar__main-info"
        >
          <div
            class="sidebar__title"
          >
            401 Unauthorized
          </div>
          <div
            class="sidebar__info"
          >
            During the Christmas season, Elliot and Mr. Robot make their return; Darlene deals with real trouble; Tyrell is bored; Dom becomes paranoid.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
